File: org-frontend/src/api/model/article.ts
// eslint-disable-next-line no-restricted-imports
import dayjs, { Dayjs } from "dayjs"

export interface Article {
  id: string
  userId: string
  title: string
  content: string
  createdAt: Dayjs
  updatedAt: Dayjs
  publishedAt: Dayjs | undefined
  published: boolean
  isEditable?: boolean
}

export function fromJson(article: Article): Article {
  return {
    ...article,
    createdAt: dayjs(article.createdAt),
    updatedAt: dayjs(article.updatedAt),
    publishedAt: article.publishedAt ? dayjs(article.publishedAt) : undefined,
  }
}

File: org-go/pkg/models/article.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Article struct {
	ID          uuid.UUID  `db:"id"`
	UserID      uuid.UUID  `db:"user_id"`
	Title       string     `db:"title"`
	Content     string     `db:"content"`
	CreatedAt   time.Time  `db:"created_at"`
	UpdatedAt   time.Time  `db:"updated_at"`
	PublishedAt *time.Time `db:"published_at"`
	Published   bool       `db:"published"`
}
